### PR TITLE
Consistent error reporting and masking provider behavior

### PR DESCRIPTION
**Pull request by jarrettranderson**

### CODE REVIEW

File: ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/AbstractMaskingProvider.java [hunk 1]
```diff
@@ -122,4 +141,43 @@ public void setName(String name) {
...
+        case MESSAGE:
+          response = unexpectedInputReturnMessage;
+          break;
+        case ERROR_EXIT:
+          throw new PrivacyProviderInvalidInputException(input, getName());
+        default:
+          response = null;
+      }
+    } else {
+      if (unspecifiedValueHandling == 2) {
```
Comment (DenisRicard) [inline ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/AbstractMaskingProvider.java:166]: Should we create constants and use those everywhere and get rid of the digits ? I see that the digits are concentrated all into this method at least so it's not as bad as before.

**Review by DenisRicard — APPROVED** [review]
Approved assuming the one small comment gets answered.
Looks good.

File: ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/AbstractMaskingProvider.java (hunk 1 above)
Comment (jarrettranderson) [reply ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/AbstractMaskingProvider.java:166]: I will create constants for the old-style numeric values as part of the next delivery for this story.

File: ipv-core/src/main/java/com/ibm/whc/deid/models/City.java [hunk 2]
```diff
@@ -23,24 +25,65 @@
...
 
     this.key = name.toUpperCase();
   }
 
+  private void init() {
+    if (name == null || name.trim().isEmpty()) {
+      throw new IllegalArgumentException(
+          Messages.getMessage(LogCodes.WPH1010E, String.valueOf(name), "city name"));
+    }
+    if (countryCode == null || countryCode.trim().isEmpty()) {
```
Comment (DenisRicard) [inline ipv-core/src/main/java/com/ibm/whc/deid/models/City.java:77]: Docs says that some columns are optional :
city
File name: city_list.csv Contains a list of cities and related information.

Mandatory column: name

Optional columns: latitude, longitude, country code, population

Comment (jarrettranderson) [reply ipv-core/src/main/java/com/ibm/whc/deid/models/City.java:77]: I will investigate and re-instate the optional nature of the documented columns as part of the next delivery for this story.

**Pull request merged**